### DISCUSSION TITLE
Recommendation for a library for .NET Dependency Injection?

### DISCUSSION THREAD

**Discussion by SaydamT110** [community/community#132345] [Packages]
### Select Topic Area

Question

### Body

Can you recommend a NuGet package to make managing the Dependency Injection process eaiser?

**Comment by TanerSaydam** [comment]
1. **Scrutor
2. Autofac
3. Ninject**

My favorite is the Scrutor.

Here is the example code for it

 ```csharp
services.Scan(action =>
        {
            action
            .FromAssemblies(Assembly.GetExecutingAssembly())
            .AddClasses(publicOnly: false)
            .UsingRegistrationStrategy(RegistrationStrategy.Skip)
            .AsMatchingInterface()
            .AsImplementedInterfaces()
            .WithScopedLifetime();
        });
```

**Reply by LiteBrite82** [reply]
Hi @TanerSaydam 

We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
